src/_interfaces/sampler.js

```diff
@@ -20,8 +20,22 @@
 // THE SOFTWARE.
 
 declare interface Sampler {
-    isSampled(operation: string): boolean;
+    name(): string;
+    
+    /**
+     * Decides if a new trace starting with given operation name
+     * should be sampled or not. If the method returns true, it
+     * must populate the tags dictionary with tags that identify
+     * the sampler, namely sampler.type and sampler.param.
+     *
+     * @param {string} operation - Operation name of the root span.
+     * @param {Object} tags - output dictionary to store sampler tags.
```

Comment:
just curious, why did you choose this approach rather than passing back tags that you can initialize once in your constructors?